checkov/github_actions/image_referencer/provider.py

```diff
@@ -45,11 +45,13 @@ def extract_images_from_workflow(self) -> list[Image]:
                 if isinstance(container, dict):
                     image = container.get("image", "")
                     start_line, end_line = container.get(START_LINE, 0), container.get(END_LINE, 0)
-
                 elif isinstance(container, str):
                     image = container
-                    start_line = [line_number for line_number, line in
-                                  self.workflow_line_numbers[job_object[START_LINE]:] if image in line][0]
+                    line_to_line_numbers = self.workflow_line_numbers[job_object[START_LINE] - 1:]
+                    lines = [line_number for line_number, line in line_to_line_numbers if image in line]
+                    if not lines:
+                        continue
+                    start_line = lines[0]
```

Comment:
since you only want the first image finding and don't need to iterate further, this should fit better
```suggestion
                    image_line = next((line_number for line_number, line in line_to_line_numbers if image in line), None)
                    if not image_line:
                        continue
                    start_line = image_line
```